src/viewport/api/public.py

```
@@ -244,11 +244,38 @@ async def _build_project_cover(
+    # Guard: the batch call may omit either key (e.g. transient S3 failure
+    # inside the loop). PublicCover requires both URLs to be non-null strings,
+    # so fall back to per-URL presigning before constructing the response.
+    # If the fallback itself fails, drop the cover rather than constructing
+    # an invalid PublicCover.
+    if full_url is None:
+        try:
+            full_url = await s3_client.generate_presigned_url_async(
+                cover_photo.object_key,
+                response_content_disposition=cover_disposition,
+            )
+        except Exception as exc:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)